### PR TITLE
Add the Vanilla Label component.

### PR DESCRIPTION
**Pull request by huwshimi**
## Done

- Add the Vanilla Label component. There is already a form Label component so this had to be StatusLabel.

### Storybook

To see rendered examples of all react-components, run:

```shell
yarn start
```

### QA in your project

from `react-components` run:

```shell
yarn build
npm pack
```

Install the resulting tarball in your project with:

```shell
yarn add <path-to-tarball>
```

### QA steps

- Visit the new StatusLabel storybook example (Edit: the demo service keeps timing out so you'll need to run this locally).
- Check that it appears like the Vanilla Label: https://vanillaframework.io/docs/patterns/labels

### CODE REVIEW

File: src/components/Button/Button.stories.mdx [hunk 1]
```diff
@@ -12,6 +12,13 @@ import Button from "./Button";
       },
       defaultValue: "Click me!",
     },
+    appearance: {
```
Comment (huwshimi) [inline src/components/Button/Button.stories.mdx:15]: Drive-by fix to show the button options in the canvas controls.

File: src/components/StatusLabel/StatusLabel.tsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
+import React from "react";
+import type { HTMLProps, ReactNode } from "react";
+
+import classNames from "classnames";
+import { ClassName, PropsWithSpread, ValueOf } from "types";
+
+export const StatusLabelAppearance = {
+  Caution: "caution",
```
Comment (bartaz) [inline src/components/StatusLabel/StatusLabel.tsx:8]: Why the camel case here? it seems in other enums we use `ALLCAPS` (like in buttons)?

Comment (huwshimi) [reply src/components/StatusLabel/StatusLabel.tsx:8]: Oh yeah sorry we've been moving to that syntax as it seems to be how it's written in the [TypeScript docs](https://www.typescriptlang.org/docs/handbook/enums.html). I'll change this to allcaps for consistency.

**Comment by huwshimi** [comment]
> The code itself looks fine, I'm just not sure about the name. `StatusLabel` is ok as a workaround, but I'm not sure we should settle on it. If we leave it like this it is going to be confusing, especially considering plans to possible merge vanilla and react-components into single project.
> 
> I guess the issue here shows that the label component in Vanilla is wrongly named, so probably we should try to rename it there and use the same name here. I'll try to bring this up on next Vanilla working group.
> 
> @huwshimi @petermakowski Is this change urgent for you, would it block you if we keep in on hold for another week or two?

This is not urgent at all, far from it, I built it in maas-ui but we didn't end up using it so I've just proposed it here so that the work doesn't get lost.

**Review by bartaz — APPROVED** [review]
LGTM
